### PR TITLE
Feature/continue return

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
Continue return functionality.
Bug fixes for returns list.

Hidden button due to work remaining - includes return status attribute and related unit tests, return status request.

### CODE REVIEW

File: src/EA.Weee.RequestHandlers/AatfReturn/GetReturnStatusHandler.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            IMapper mapper)
+        {
+            this.authorization = authorization;
+            this.returnDataAccess = returnDataAccess;
+            this.mapper = mapper;
+        }
+
+        public async Task<ReturnStatus> HandleAsync(GetReturnStatus message)
+        {
+            authorization.EnsureCanAccessExternalArea();
```
Comment (SamGoodeySFW) [inline src/EA.Weee.RequestHandlers/AatfReturn/GetReturnStatusHandler.cs]: Had a look in the submit return handler tests and couldn't see tests to cover this handler e.g. ensure user access external area, making sure the error is thrown when @return is null etc.

File: src/EA.Weee.Web/Areas/AatfReturn/Controllers/ReturnsController.cs [hunk 2]
```diff
@@ -66,5 +66,11 @@ private async Task SetBreadcrumb(Guid organisationId, string activity)
             breadcrumb.ExternalActivity = activity;
             breadcrumb.SchemeInfo = await cache.FetchSchemePublicInfo(organisationId);
         }
+
+        [HttpGet]
+        public virtual async Task<ActionResult> Continue(Guid organisationId, Guid returnId, bool redirectReportingOptions, bool redirectSelectYourPcs, bool redirectTaskLis)
+        {
+            return AatfRedirect.SelectReportOptions(organisationId, returnId);
```
Comment (SamGoodeySFW) [inline src/EA.Weee.Web/Areas/AatfReturn/Controllers/ReturnsController.cs]: Unit test to make sure the user is redirected?

File: src/EA.Weee.RequestHandlers/AatfReturn/GetReturnStatusHandler.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+{
+    using System;
+    using System.Threading.Tasks;
+    using Prsd.Core.Mapper;
+    using Prsd.Core.Mediator;
+    using Requests.AatfReturn;
+    using Security;
+    using ReturnStatus = Core.AatfReturn.ReturnStatus;
+
+    internal class GetReturnStatusHandler : IRequestHandler<GetReturnStatus, ReturnStatus>
```
Comment (JB12298) [inline src/EA.Weee.RequestHandlers/AatfReturn/GetReturnStatusHandler.cs]: Need to add unit tests

File: src/EA.Weee.RequestHandlers/Mappings/ReturnMap.cs [hunk 4]
```diff
@@ -35,7 +37,8 @@ public ReturnData Map(ReturnQuarterWindow source)
                 CreatedDate = source.Return.CreatedDate,
```
Comment (JB12298) [inline src/EA.Weee.RequestHandlers/Mappings/ReturnMap.cs]: Nothing you've changed, but apparently SchemeDataItems has no unit test attached to it?

File: src/EA.Weee.Web/Areas/AatfReturn/Attributes/ValidateReturnActionFilterAttribute.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            if (!(Guid.TryParse(returnIdActionParameter.ToString(), out var guidReturnIdActionParameter)))
+            {
+                throw new ArgumentException("The specified return ID is not valid.");
+            }
+
+            var returnId = (Guid)guidReturnIdActionParameter;
+
+            using (var client = Client())
+            {
+                var @returnStatus = client.SendAsync(context.HttpContext.User.GetAccessToken(), new GetReturnStatus(returnId));
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/AatfReturn/Attributes/ValidateReturnActionFilterAttribute.cs]: Should there be a unit test covering this client call?

File: src/EA.Weee.RequestHandlers/Mappings/ReturnMap.cs (hunk 4 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers/Mappings/ReturnMap.cs]: Added unit test for this property

File: src/EA.Weee.RequestHandlers/AatfReturn/GetReturnStatusHandler.cs (hunk 1 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers/AatfReturn/GetReturnStatusHandler.cs]: Added unit tests

File: src/EA.Weee.Web/Areas/AatfReturn/Controllers/ReturnsController.cs (hunk 2 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/AatfReturn/Controllers/ReturnsController.cs]: Changed the way this works.

File: src/EA.Weee.RequestHandlers/AatfReturn/GetReturnStatusHandler.cs (hunk 3 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers/AatfReturn/GetReturnStatusHandler.cs]: Added unit tests

File: src/EA.Weee.Web/Areas/AatfReturn/Attributes/ValidateReturnActionFilterAttribute.cs (hunk 5 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/AatfReturn/Attributes/ValidateReturnActionFilterAttribute.cs]: Unit tests added

**Pull request merged**